### ISSUE TITLE
argocd-util missing repository TLS client and insecure export details

### ISSUE THREAD

**Issue by alexec** [argoproj/argo-cd#2075]
**Describe the bug**

the `getReferenceSecrets` method does not have some of the new TLS client fields. Probably should look like this:

```golang
	if reposRAW, ok := cm.Data["repositories"]; ok {
		repoCreds := make([]settings.RepoCredentials, 0)
		err := yaml.Unmarshal([]byte(reposRAW), &repoCreds)
		errors.CheckError(err)
		for _, cred := range repoCreds {
			if cred.PasswordSecret != nil {
				referencedSecrets[cred.PasswordSecret.Name] = true
			}
			if cred.SSHPrivateKeySecret != nil {
				referencedSecrets[cred.SSHPrivateKeySecret.Name] = true
			}
			if cred.UsernameSecret != nil {
				referencedSecrets[cred.UsernameSecret.Name] = true
			}
			if cred.TLSClientCASecret != nil {
				referencedSecrets[cred.TLSClientCASecret.Name] = true
			}
			if cred.TLSClientCertKeySecret != nil {
				referencedSecrets[cred.TLSClientCertKeySecret.Name] = true
			}
			if cred.TLSClientCertDataSecret != nil {
				referencedSecrets[cred.TLSClientCertDataSecret.Name] = true
			}
		}
	}
```

**Comment by alexec** [comment]
@jannfis FYI

**Comment by alexmt** [comment]
Nice catch. Just to add some info. `argocd_util` is used to backup/restore argocd settings: https://argoproj.github.io/argo-cd/operator-manual/disaster_recovery/

**Comment by jannfis** [comment]
Indeed, good catch. I have fixed that in #2098, and argocd-util also exports the custom CAs and SSH known hosts now. Didn't even know about the existence of argocd-util until now :)
